components/logins/src/error.rs

```diff
@@ -102,7 +98,170 @@ impl Error {
                 InvalidLogin::NoTarget => "InvalidLogin::NoTarget",
                 InvalidLogin::IllegalFieldValue { .. } => "InvalidLogin::IllegalFieldValue",
             },
-            ErrorKind::ProtobufDecodeError(_) => "BufDecodeError",
+        }
+    }
+}
+
+// TODO: Move the above to somewhere else - everything below is our "public" error list.
+// Probably copy the pattern started by fxa-client - everything above moves to `internal/`?
+// Originally exposed manually via LoginsStorageException.kt.
+
+#[derive(Debug, thiserror::Error)]
+pub enum LoginsError {
+    // In the .kt, this was actually the base-class and lots of Android code
+    // caught this exception when they want *any* exception.
+    // To confuse further, it was also thrown as a generic 'unexpected' exception.
+    // For now, let's keep those things distinct, but for parity with the old
+    // code we should see if we can replace Unexpected with whatever base-class
+    // we end up with.
+    // XXX - TODO - work out the base-class story so the above remains true.
```

Comment:
Just a reminder to look into this TODO - I really have no idea if the errors generated for kotlin by uniffi will be able to help us here - probably not a huge deal if not, but it will impact the changes in the a-c repo as that might be where the class hierarchy for exceptions is re-established)